controllers/dscinitialization/utils.go

```diff
@@ -18,132 +19,184 @@ import (
 	"k8s.io/client-go/util/retry"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 
 	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
 )
 
 var (
 	resourceInterval = 10 * time.Second
 	resourceTimeout  = 1 * time.Minute
 )
 
-// createOdhNamespace creates a Namespace with given name and with ODH defaults. The defaults include:
-// - Odh specific labels
-// - Pod security labels for baseline permissions
-// - ConfigMap  'odh-common-config'
-// - Network Policies 'opendatahub' that allow traffic between the ODH namespaces
-// - RoleBinding 'opendatahub'.
-func (r *DSCInitializationReconciler) createOdhNamespace(ctx context.Context, dscInit *dsciv1.DSCInitialization, name string, platform cluster.Platform) error {
+// createOperatorResource include steps:
+// - 1. validate customized application namespace || create/update default application namespace
+// - 2. patch application namespaces for managed cluster
+//   - Odh specific labels for access
+//   - Pod security labels for baseline permissions
+//
+// - 3. Patch monitoring namespace
+// - 4. Network Policies 'opendatahub' that allow traffic between the ODH namespaces
+// - 5. ConfigMap  'odh-common-config'
+// - 6. RoleBinding 'opendatahub'.
+func (r *DSCInitializationReconciler) createOperatorResource(ctx context.Context, dscInit *dsciv1.DSCInitialization, nsName string, platform cluster.Platform) error {
```

Comment:
is it required to pass the `nsName` here since there is already a `DSCInitialization` parameter ? According to my understanding, the `nsName` is always set to `instance.Spec.ApplicationsNamespace`

Reply:
i did not change this from old implementation. but i can get namespace variable